src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit.cpp

```diff
@@ -179,54 +181,79 @@ Pass Orbit::getPassWithRevolutionNumber(const Integer& aRevolutionNumber, const
 
     const std::lock_guard<std::mutex> lock {this->mutex_};
 
-    const auto passMapIt = this->passMap_.find(aRevolutionNumber);
-
-    if (passMapIt != this->passMap_.end())
-    {
-        return passMapIt->second;
-    }
-    else
+    const auto getClosestPass = [&passMap_](const Integer& aRevolutionNumber) -> Pass
     {
-        // If any, get pass with closest revolution number
+        if (this->passMap_.empty())
+        {
+            return Pass::Undefined();
+        }
 
-        Pass const* closestPasSPtr = nullptr;
+        // exact revolution number exists
+
+        if (this->passMap_.count(aRevolutionNumber))
+        {
+            return this->passMap_.at(aRevolutionNumber);
+        }
 
         const auto lowerBoundMapIt = this->passMap_.lower_bound(aRevolutionNumber);
 
-        if (lowerBoundMapIt != this->passMap_.end())
-        {
-            if (lowerBoundMapIt == this->passMap_.begin())
-            {
-                closestPasSPtr = &(lowerBoundMapIt->second);
-            }
-            else
-            {
-                const auto closestPassMapIt = std::prev(lowerBoundMapIt);
+        // Revolution number is greater than any existing revolution number in map
+        // {5, 6, 9, 10} -> aRevolutionNumber=12 -> return 10
 
-                if ((aRevolutionNumber - closestPassMapIt->first) < (lowerBoundMapIt->first - aRevolutionNumber))
-                {
-                    closestPasSPtr = &(closestPassMapIt->second);
-                }
-                else
-                {
-                    closestPasSPtr = &(lowerBoundMapIt->second);
-                }
-            }
+        if (lowerBoundMapIt == this->passMap_.end())
+        {
+            return this->passMap_.rbegin()->second;
         }
-        else if (this->passMap_.size() > 0)
+
+        // Revolution number is lesser than any existing revolution number in map
+        // {5, 6, 9, 10} -> aRevolutionNumber=4 -> return 5
+
+        if (lowerBoundMapIt == this->passMap_.begin())
         {
-            closestPasSPtr = &(this->passMap_.begin()->second);
+            return this->passMap_.begin()->second;
         }
 
-        Pass currentPass = Pass::Undefined();
+        // Cloases revolution number is within the map
+
+        const auto closestPassMapIt = std::prev(lowerBoundMapIt);
 
-        if (closestPasSPtr != nullptr)  // Pass with closest revolution number found
+        // {5, 6, 9, 10} -> aRevolutionNumber=8 -> return 9
+        // lowerBoundMapIt = 9, closestPassMapIt = 6
```

Comment:
```suggestion
        // {5, 6, 9, 10} -> aRevolutionNumber=7 -> return 6
        // lowerBoundMapIt = 9, closestPassMapIt = 6
```

Is this comment supposed to be switched with the other one? Since `closestPassMapIt` is the one before the revolution number

Feel free to close these if I'm just confused